auxx/Main.hs

```diff
@@ -69,30 +80,34 @@ runNodeWithSinglePlugin ::
 runNodeWithSinglePlugin nr (plugin, plOuts) =
     runNode nr ([plugin], plOuts)
 
-action :: HasCompileInfo => AuxxOptions -> Either (WithCommandAction AuxxMode) Text -> Production ()
-action opts@AuxxOptions {..} command = withConfigurations conf $ do
+action :: HasCompileInfo => AuxxOptions -> Either WithCommandAction Text -> Production ()
+action opts@AuxxOptions {..} command = do
     CLI.printFlags
-    logInfo $ sformat ("System start time is "%shown) $ gdStartTime genesisData
-    t <- currentTime
-    logInfo $ sformat ("Current time is "%shown) (Timestamp t)
-    (nodeParams, tempDbUsed) <-
-        correctNodeParams opts =<< CLI.getNodeParams cArgs nArgs
     let
-        toRealMode :: AuxxMode a -> RealMode EmptyMempoolExt a
-        toRealMode auxxAction = do
-            realModeContext <- ask
-            let auxxContext =
-                    AuxxContext
-                    { acRealModeContext = realModeContext
-                    , acCmdCtx = cmdCtx
-                    , acTempDbUsed = tempDbUsed }
-            lift $ runReaderT auxxAction auxxContext
-    let vssSK = unsafeFromJust $ npUserSecret nodeParams ^. usVss
-    let sscParams = CLI.gtSscParams cArgs vssSK (npBehaviorConfig nodeParams)
-    bracketNodeResources nodeParams sscParams txpGlobalSettings initNodeDBs $ \nr ->
-        runRealBasedMode toRealMode realModeToAuxx nr $
-            (if aoNodeEnabled then runNodeWithSinglePlugin nr else identity)
-            (auxxPlugin opts command)
+        runWithoutNode = do
+          rawExec Nothing Nothing opts Nothing command
+    (flip catch) (\(_ :: SomeException) -> runWithoutNode) $ withConfigurations conf $ do
```

Comment:
1. `flip catch` is called `handle`
2. don't catch `SomeException`, it could be (1) an async exception, (2) a fatal error

and it looks like `auxxPlugin` itself is within this `catch`: does this mean that if I run with `node` enabled, then fail with an exception during command execution, your handler will catch it and run without node?

I suggest that you don't rely on exceptions when determining whether to run with node or without.

Reply:
> (1) an async exception

It's wrong, `catch` from `Universum` doesn't catch async exceptions. Other than that, I agree with
> I suggest that you don't rely on exceptions when determining whether to run with node or without.